coap-gateway/service/resourceDirectory.go

```diff
@@ -91,13 +90,9 @@ func resourceDirectoryPublishHandler(req *mux.Message, client *Client) {
 		return
 	}
 
-	for _, resource := range publishedResources {
-
-		err := client.observeResource(req.Context, resource.GetResourceID(), resource.IsObservable(), true)
-		if err != nil {
-			log.Errorf("unable to start observation of %v", resource.GetResourceID(), err)
-		}
-	}
+	client.server.taskQueue.Submit(func() {
```

Comment:
we need to leave it is and plan to remove it when it will be fixed in device.
https://github.com/iotivity/iotivity-lite/blob/311705219fd4c17fb5dc858a386e051284c14a80/api/cloud/rd_client.c#L69

Reply:
Are we going to fix it in iotivity-lite? Can you please create a bug in their github ?
Thanks